### ISSUE TITLE
Overhaul ANSI & Base16 Colours

### ISSUE THREAD

**Issue by justTOBBI** [catppuccin/catppuccin#1961]
A person on [twitter](https://fxtwitter.com/phaazon_/status/1631605104829710336?t=Ta06KoK1JBgCNTGMxOxsEg&s=19) pointed out, that the term colors are broken due to them not having a light variant. We need ctp 2.1 to have more term colors. 
![image](https://user-images.githubusercontent.com/47004621/222765646-cab9a16f-cb36-4725-8a2c-4f11316c7ce3.png)

**Comment by backwardspy** [comment]
i know @extremepayne was looking into this a little while back, where did you get to with it?

**Comment by nekowinston** [comment]
Yeah, I've been complaining about this issue too since June last year. I think it's fair to say its a known issue, and something we want to address. It will impact other ports though, not just terminal emulators.

Ports that come to mind:
- Every terminal emulator, of course
- IDEs/code editors with a Terminal function (JetBrains/VSCode, etc.)
- TUI/CLI without truecolor support
  - Newsboat
  - GitUI
  - mc

I'm pretty sure there are more CLI tools that lack truecolor support, but those 3 are the first that come to mind.

**Comment by extremepayne** [comment]
> i know @extremepayne was looking into this a little while back, where did you get to with it?

Sorry for missing this ping; got lost in my inbox. I do have a set of colors I decided to use and I recorded them all in [my fork of the alacritty port](https://github.com/extremepayne/catppuccin-alacritty) (chosen because I find the config format readable; i don't actually use alacritty). I'll put the macchiato colors (commented) here.

```yaml
    # Normal colors
    normal:
        black: "#494D64" # surface1
        red: "#ED8796" # red
        green: "#A6DA95" # green
        yellow: "#F5A97F" # peach
        blue: "#8AADF4" # blue
        magenta: "#C6A0F6" # muave
        cyan: "#8BD5CA" # teal
        white: "#B8C0E0" # subtext1

    # Bright colors
    bright:
        black: "#5B6078" # surface2
        red: "#EE99A0" # maroon
        green: "#ADD9A8" # green but bright
        yellow: "#EED49F" # yellow
        blue: "#7DC4E4" # sapphire
        magenta: "#F5BDE6" # pink
        cyan: "#91D7E3" # sky
        white: "#A5ADCB" # subtext0
```
This hits most of the colors and keeps to the names they are given as part of the sixteen. It's missing rosewater, flamingo, and lavender, and it has a modified green simply because ctp has more reddish-purple and fewer green colors than the ANSI 16. Y'all would know better than me which colors are important to represent, I chose the ones I thought looked best. I'm also not certain if I've chosen the right direction for which ones are bright (peach and mauve certainly have more impact than yellow and pink, imo, so should those be bright?). But overall I think the pairings make sense and bring some differentiation to the terminal. I've been using this for a couple weeks and think it works great.

**Comment by MrDowntempo** [comment]
Is this something [Base24](https://github.com/Base24/Base24.github.io) could fix? Which is a variant of Base16 color theme designer which adds 8 more colors for ANSI compliance.

**Comment by justTOBBI** [comment]
@MrDowntempo thanks for the suggestion. It looks like it could very well fit our purpose. Looking into it, do you, by any chance, know what file format the custom theme, in this case catppuccin, has to be in?

**Comment by MrDowntempo** [comment]
> @MrDowntempo thanks for the suggestion. It looks like it could very well fit our purpose. Looking into it, do you, by any chance, know what file format the custom theme, in this case catppuccin, has to be in?

I am not 100% sure, but I _thiiiiiiiiink_ it should be compatible with the Base16 format you already have available:
https://github.com/catppuccin/base16

**Comment by justTOBBI** [comment]
Maybe someone looking for a meaningful contribution to the Hacktoberfest could set its eyes on this?
Should I change the title to say  HACKTOBERFEST: [title]?

**Comment by nekowinston** [comment]
> Maybe someone looking for a meaningful contribution to the Hacktoberfest could set its eyes on this? Should I change the title to say HACKTOBERFEST: [title]?

Since this impacts more than two dozen ports, I don't think this is up for grabs for Hacktoberfest. It's on our radar for the next update of the Catppuccin palette.

**Comment by justTOBBI** [comment]
![image](https://github.com/catppuccin/catppuccin/assets/47004621/e452c563-1880-46c5-b16f-74fba5f42c89)
I experimented with [terminal.sexy](https://terminal.sexy/#MDRGxtD1YmiA54KEptGJ5ciQjKru9Ljkgci-pa3Ob2-A6JmZuNGf5s2opLjt9dDpk8e6usHP) and got some pretty decent results when I light up the colours by ~10 points (Click on the colour and then subtract `10` from the `S` value)

Let me know how you feel about this :)

**Comment by Maganoos** [comment]
code for kitty :3 - using both official palette and https://www.w3schools.com/colors/colors_picker.asp :3
```
# The 16 terminal colors

# black
color0 #494d64
color8 #5b6078

# red
color1 #ed8796
color9 #ee99a0

# green
color2  #a6da95
color10 #c0e4b4

# yellow
color3  #f5a97f
color11 #eed49f

# blue
color4  #8aadf4
color12 #b9cef8

# magenta
color5  #ee90d5
color13 #f5bde6

# cyan
color6  #8bd5ca
color14 #b4e4dd

# white
color7  #b8c0e0
color15 #a5adcb
```

**Comment by sgoudham** [comment]
Hey :wave:, it's about time that we got around to this. @catppuccin/staff and @catppuccin/core have been working on redefining the ANSI colours with the goal of defining different colours for the brights versus the normals.

We've come up with the following (just showcasing Latte and Mocha for convenience sake as Frappé and Macchiato won't be far behind Mocha):

**Latte**:

![image](https://github.com/user-attachments/assets/b2cc1ed3-6bc3-4957-803d-fb5cba3fd3e8)
![image](https://github.com/user-attachments/assets/63f2c659-163e-4101-aff3-52f133812114)

**Mocha**:

![image](https://github.com/user-attachments/assets/fb199fa7-ef6f-4628-8d5c-07004b259e2c)
![image](https://github.com/user-attachments/assets/a548f4d7-ed50-4dfa-9114-0c8ab1ffe2eb)

What's interesting is that Catppuccin, as a pastel colour scheme and with how the colours are defined, has a deep "incompatibility" with the ANSI standard. The darker flavours (Frappé, Macchiato & Mocha) struggle with this a lot more than Latte as placing our accent colours on top of one other is not recommended and using `white` on top of accent colours is not recommended either.

We have a tough decision to make where I believe we have 3 choices: 

1. Break the ANSI standard and use a dark colour in place of `white` and a light colour in place of `black` to try and improve the look across applications. (This is ugly because it means when people specify `white`, they will see a dark colour but it should _technically_ improve readability)
2. We stop being "pastel" by forcing different colours that inherently have higher contrast with each other. 
3. We accept the somewhat harsh reality that TUIs simply aren't going to look that great out of the box (unless created with Catppuccin in mind by specifying `black` to act as the "text" colour or providing a way to customise the TUI outside of ANSI)

Personally, I'm okay with No. 3 as the primary goal of our work here was to separate the normal colours from the brights, which I think we have done while keeping to the pastel nature of the theme.

**Comment by sgoudham** [comment]
An update regarding this issue, we continued to iterate on the new ANSI colours and have implemented them in https://github.com/catppuccin/palette/pull/98.

We wanted to ensure that the `normal` colours were default Catppuccin colours to retain the look and feel of a pastel colour scheme and we've made sure that `black` and `white` are ordered (i.e. bright black is brighter than normal black and bright white is brighter than normal white)

See below for screenshots of the new ANSI colours:

`Latte`
![image](https://github.com/user-attachments/assets/a69f2337-dc50-4d7b-adf9-2aca7748c144)

`Frappe`
![image](https://github.com/user-attachments/assets/d14d4d3e-6b4c-49d6-bf59-a39a4205a78d)

`Macchiato`
![image](https://github.com/user-attachments/assets/897f2056-d207-4a0f-8721-ed6eff57b25c)

`Mocha`
![image](https://github.com/user-attachments/assets/a2e3935d-2965-4c4a-bddf-1243c2462ec4)

Ultimately, the decision that we've made is to go with 3. as mentioned before where the colours follow the ANSI standard but, depending on how the application uses the ANSI colours, readability may suffer. When @unseen-ninja and I were testing these colours, we tried making the `white` traditionally more white and the `black` traditionally more black but we found that it ended up looking good for one application but made it worse for an application that already looked good. 

I've listed some examples below of some parts of applications with the old ANSI colours versus the new ones. It's tough to call them improvements but I think this is the best we can do if we're to stay as a pastel theme while properly adhering to ANSI standards:

<details>
<summary>dive</summary>

`Latte Old`
![image](https://github.com/user-attachments/assets/ad6a3c39-7573-4e48-8745-589950028cdb)

`Latte New`
![image](https://github.com/user-attachments/assets/be01c5df-ddd3-4323-b422-caf9e8de2c9b)

`Mocha Old`
![image](https://github.com/user-attachments/assets/83f81975-eea6-4f4e-b0b2-c5ffa2f9bc99)

`Mocha New`
![image](https://github.com/user-attachments/assets/3a493460-e164-459e-82a4-7051aa51b65b)

</details>

<details>
<summary>gitui</summary>

`Latte Old`
![image](https://github.com/user-attachments/assets/736efbc4-e882-46d2-b557-4aa46128edd8)

`Latte New`
![image](https://github.com/user-attachments/assets/b38269c5-cdc0-4391-ad46-0657e1d73c9c)

`Mocha Old`
![image](https://github.com/user-attachments/assets/e8136594-7802-4b67-ad5b-7b1cac30ad4d)

`Mocha New`
![image](https://github.com/user-attachments/assets/aaab16ed-ac55-4c9e-af9c-e85e017a79d3)

</details>

The plan is for https://github.com/catppuccin/palette/pull/98 to be merged along with #2578 and have those updates filter down to our existing palette ports, specifically [catppuccin/rust](https://github.com/catppuccin/rust) where then it will be included in [catppuccin/whiskers](https://github.com/catppuccin/whiskers) and finally then propagated to our terminal ports via whiskers templates. We'll also need to do some additional work in propagating these ANSI colour updates to various places upstream. 

I'll be closing this issue as resolved soon as I believe the base work has been carried out. I'll create a project board to track the progress on updating all of our different terminals and I'll create a separate issue on [catppuccin/base16](https://github.com/catppuccin/base16) to track improvements to our base16 offering.

Thanks everyone for their input on this issue.

**Comment by sgoudham** [comment]
Created the project board in https://github.com/orgs/catppuccin/projects/22, worth noting that this work can only start after a whiskers version has been published including the ANSI colours.
